libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java

```diff
@@ -1452,4 +1467,89 @@ public static String markMediaFailed(Context context, @NonNull String postConten
         }
         return postContent;
     }
+
+    public static boolean hasMediaItemsMarkedUploading(Context context, @NonNull String postContent) {
+        return hasMediaItemsMarkedWithTag(context, postContent, ATTR_STATUS_UPLOADING);
+    }
+
+    public static boolean hasMediaItemsMarkedFailed(Context context, @NonNull String postContent) {
+        return hasMediaItemsMarkedWithTag(context, postContent, ATTR_STATUS_FAILED);
+    }
+
+    private static boolean hasMediaItemsMarkedWithTag(Context context, @NonNull String postContent, String tag) {
+        // fill in Aztec with the post's content
+        AztecText content = new AztecText(context);
```

Comment:
ok I tried and measured the amount of time spent in the for loop versus using `content.getAllElementAttributes()` with the predicate, for a Post containing more than 50 images, both approaches resulted in very similar times, approximately 0 to 1 millisecond spent most of the times, with a 3 ms outlier (but still a non-significant value):

```
08-08 11:36:02.040 15486-15486/org.wordpress.android E/WordPress-PROFILING: TOTAL TIME: 1
08-08 11:36:02.074 15486-15486/org.wordpress.android E/WordPress-PROFILING: TOTAL TIME: 1
08-08 11:36:02.214 15486-15486/org.wordpress.android E/WordPress-PROFILING: TOTAL TIME: 0
08-08 11:39:36.475 19524-19524/org.wordpress.android E/WordPress-PROFILING: TOTAL TIME: 3
08-08 11:39:36.795 19524-19524/org.wordpress.android E/WordPress-PROFILING: TOTAL TIME: 1
```

This is using the change proposed, measuring time before and after the code block containing both the `getSpans` call and the `for` loop to iterate through them:

```
08-08 11:42:01.968 20960-20960/org.wordpress.android E/WordPress-PROFILING: TOTAL TIME: 0
08-08 11:42:02.052 20960-20960/org.wordpress.android E/WordPress-PROFILING: TOTAL TIME: 1
08-08 11:48:24.001 20960-20960/org.wordpress.android E/WordPress-PROFILING: TOTAL TIME: 0
08-08 11:48:24.049 20960-20960/org.wordpress.android E/WordPress-PROFILING: TOTAL TIME: 0
```

I think given in both cases the time is very similar and, in any case, not significant, I'd prefer to go with the code that uses predicates as for having a separation, i.e. so we don't need to look within the spans contained by the `AztecText`  component but rather using the exposed API.